hc5-async-json/src/test/java/com/ok2c/hc5/json/http/JsonResponseConsumerTest.java

```diff
@@ -0,0 +1,150 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JsonResponseConsumerTest {
+
+    static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Test
+    void testNonJsonErrorResponseBodyIgnored() throws Exception {
+        String errorBody = "Unexpected internal failure";
+
+        AsyncResponseConsumer<Message<HttpResponse, RequestData>> consumer =
+                JsonResponseConsumers.create(objectMapper, RequestData.class);
+
+        CompletableFuture<RequestData> resultFuture = new CompletableFuture<>();
+        consumer.consumeResponse(
+                new BasicHttpResponse(HttpStatus.SC_INTERNAL_SERVER_ERROR),
+                new BasicEntityDetails(errorBody.length(), ContentType.TEXT_PLAIN),
+                HttpClientContext.create(),
+                new FutureCallback<Message<HttpResponse, RequestData>>() {
+                    @Override
+                    public void completed(Message<HttpResponse, RequestData> result) {
+                        // NOTE: The error body is lost, we would need a custom consumer to retrieve that.
+                        handleResponseResult(result, resultFuture);
+                    }
+
+                    @Override
+                    public void failed(Exception ex) {
+                        resultFuture.completeExceptionally(ex);
+                    }
+
+                    @Override
+                    public void cancelled() {
+                        resultFuture.cancel(false);
+                    }
+                });
+        consumer.consume(ByteBuffer.wrap(errorBody.getBytes(StandardCharsets.UTF_8)));
+        consumer.streamEnd(null);
+
+        // SUCCESS: Non-JSON content-type is ignored by the consumer, it returns a 'null' to the callback.
+        assertThat(resultFuture).hasFailedWithThrowableThat()
+                .isInstanceOf(HttpResponseException.class);
+    }
+
+    @Test
+    void testJsonErrorResponseBodyNotMatchingResponseClass() throws Exception {
+        String errorBody = "{\"code\": 500, \"message\": \"Unexpected internal failure\"}";
+
+        AsyncResponseConsumer<Message<HttpResponse, RequestData>> consumer =
+                JsonResponseConsumers.create(objectMapper, RequestData.class);
+
+        CompletableFuture<RequestData> resultFuture = new CompletableFuture<>();
+        consumer.consumeResponse(
+                new BasicHttpResponse(HttpStatus.SC_INTERNAL_SERVER_ERROR),
+                new BasicEntityDetails(errorBody.length(), ContentType.APPLICATION_JSON),
+                HttpClientContext.create(),
+                new FutureCallback<Message<HttpResponse, RequestData>>() {
+                    @Override
+                    public void completed(Message<HttpResponse, RequestData> result) {
+                        // NOTE: The error body is lost, we would need a custom consumer to retrieve that. Actually
+                        // it might also be mapped accidentally to the response class.
```

Comment:
this is also does not apply anymore, we cannot accidentally map the error body to the response class, because now it is being filtered out.

```suggestion
                        // NOTE: The error body is lost, we would need a custom consumer to retrieve that.
```